app.hypermedia.testing.dsl/src/main/java/app/hypermedia/testing/dsl/validation/HydraValidator.xtend

```diff
@@ -3,23 +3,75 @@
  */
 package app.hypermedia.testing.dsl.validation
 
+import app.hypermedia.testing.dsl.hydra.HydraPackage
+import org.eclipse.xtext.validation.Check
+import java.net.URI
+import app.hypermedia.testing.dsl.hydra.NamespaceDeclaration
+import app.hypermedia.testing.dsl.core.CorePackage
+import java.net.URISyntaxException
+import app.hypermedia.testing.dsl.core.ClassBlock
+import app.hypermedia.testing.dsl.hydra.OperationBlock
+import app.hypermedia.testing.dsl.hydra.RelaxedOperationBlock
+import app.hypermedia.testing.dsl.hydra.PrefixedName
 
 /**
  * This class contains custom validation rules. 
  *
  * See https://www.eclipse.org/Xtext/documentation/303_runtime_concepts.html#validation
  */
 class HydraValidator extends AbstractHydraValidator {
-	
-//	public static val INVALID_NAME = 'invalidName'
-//
-//	@Check
-//	def checkGreetingStartsWithCapital(Greeting greeting) {
-//		if (!Character.isUpperCase(greeting.name.charAt(0))) {
-//			warning('Name should start with a capital', 
-//					HydraPackage.Literals.GREETING__NAME,
-//					INVALID_NAME)
-//		}
-//	}
-	
+    static final String INVALID_URI = 'Value is not a valid URI'
+    
+    @Check
+    def checkValidUri(NamespaceDeclaration ns) {
+        if (tryParseUri(ns.namespace) === false) {
+            error(INVALID_URI,
+                  HydraPackage.Literals.NAMESPACE_DECLARATION__NAMESPACE)
+        }
+    }
+    
+    @Check
+    def checkValidUri(ClassBlock clas) {
+        if (tryParseUri(clas.name.value) === false) {
+            error(INVALID_URI,
+                  CorePackage.Literals.CLASS_BLOCK__NAME
+            )
+        }
+    }
+    
+    @Check
+    def checkValidUri(OperationBlock clas) {
+        if (tryParseUri(clas.name.value) === false) {
+            error(INVALID_URI,
+                  HydraPackage.Literals.OPERATION_BLOCK__NAME
+            )
+        }
+    }
+    
+    @Check
+    def checkValidUri(RelaxedOperationBlock clas) {
+        if (tryParseUri(clas.name.value) === false) {
+            error(INVALID_URI,
+                  HydraPackage.Literals.RELAXED_OPERATION_BLOCK__NAME
+            )
+        }
+    }
+    
+    @Check
+    def checkNamespace(NamespaceDeclaration it) {
+        if (!namespace.endsWith('#') && !namespace.endsWith('/') && !namespace.endsWith(':')) {
+            warning("Namespaces usually end with hash (#), slash (/) or colon (:) characters",
+                    HydraPackage.Literals.NAMESPACE_DECLARATION__NAMESPACE
+            )
+        }
+    }
+    
+    private def tryParseUri(String uri) {
+        try {
+            val parsed = new URI(uri)
+            return parsed.scheme !== null
```

Comment:
This baffles me. Somehow a string like `12345` gets parsed successfully by `new URI`. 

Know of a better way to check URI for validity @nicky508 ?